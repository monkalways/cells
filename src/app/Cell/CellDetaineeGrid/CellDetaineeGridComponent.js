import React from 'react';
import PropTypes from 'prop-types';
import { Grid, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

const CellDetaineeGridComponent = ({ classes, children }) => (
  <Grid container className={classes.container} spacing={8} justify="center">
    {children}
  </Grid>
);

CellDetaineeGridComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  container: {
    height: theme.spacing.unit * 97,
    overflowY: 'auto',
    msOverflowStyle: '-ms-autohiding-scrollbar',
    marginLeft: 0,
    marginRight: 0,
    marginBottom: theme.spacing.unit * 0.4,
    backgroundColor: theme.palette.background.default,
    width: '100%',
  },
}))(CellDetaineeGridComponent);
