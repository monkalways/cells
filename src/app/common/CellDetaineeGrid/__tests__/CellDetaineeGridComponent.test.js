import React from 'react';
import { shallow } from 'enzyme';

import { CellDetaineeGridComponent } from '../CellDetaineeGridComponent';

describe('CellDetaineeGridComponent', () => {
  let classes;
  let children;

  beforeEach(() => {
    classes = {
      container: 'container',
    };
    children = <div>Hello Test</div>;
  });

  const setup = () => shallow(<CellDetaineeGridComponent classes={classes} children={children} />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
