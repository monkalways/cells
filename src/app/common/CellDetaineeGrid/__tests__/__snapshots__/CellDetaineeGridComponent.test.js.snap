// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellDetaineeGridComponent should render 1`] = `
<WithStyles(Grid)
  className="container"
  container={true}
  justify="center"
  spacing={8}
>
  <div>
    Hello Test
  </div>
</WithStyles(Grid)>
`;
